NEMO/apps/kiosk/views.py

```diff
@@ -433,3 +449,159 @@ def get_badge_reader(request) -> BadgeReader:
+        )
+        return render(request, "kiosk/tool_report_problem.html", dictionary)
+
+    if task.force_shutdown:
+        # Shut down the tool.
+        task.tool.operational = False
+        task.tool.save()
+        # End any usage events in progress for the tool or the tool's children.
+        UsageEvent.objects.filter(tool_id__in=task.tool.get_family_tool_ids(), end=None).update(end=timezone.now())
+        # Lock the interlock for this tool.
+        try:
+            tool_interlock = Interlock.objects.get(tool__id=task.tool.id)
+            tool_interlock.lock()
+        except Interlock.DoesNotExist:
+            pass
+
+    if task.safety_hazard:
+        concern = (
+            "This safety issue was automatically created because a "
+            + str(task.tool).lower()
+            + " problem was identified as a safety hazard.\n\n"
+        )
+        concern += task.problem_description
+        issue = SafetyIssue.objects.create(reporter=request.user, location=task.tool.location, concern=concern)
+        send_safety_email_notification(request, issue)
+
+    send_new_task_emails(request, task, None)
+    set_task_status(request, task, request.POST.get("status"), request.user)
+
+    return redirect("kiosk_tool_information", tool_id=tool.id, user_id=customer.id, back=back)
+
+
+@login_required
+@permission_required("NEMO.kiosk")
+@require_POST
+def tool_post_comment(request, tool_id, user_id, back):
+    tool = Tool.objects.get(id=tool_id, visible=True)
+    customer = User.objects.get(id=user_id)
+
+    dictionary = {
+        "tool": tool,
+        "date": None,
+        "customer": customer,
+        "back": back,
+    }
+
+    return render(request, "kiosk/tool_post_comment.html", dictionary)
+
+
+@permission_required("NEMO.kiosk")
+@require_POST
+def post_comment(request):
+    tool = Tool.objects.get(id=request.POST["tool"])
+    customer = User.objects.get(id=request.POST["customer_id"])
+    back = request.POST["back"]
+
+    dictionary = {"back": back, "tool": tool, "customer": customer}
+
+    """ Post a comment for a tool. """
+    form = CommentForm(request.POST)
+    if not form.is_valid():
+        dictionary["message"] = nice_errors(form).as_ul()
+        if "content" in form.cleaned_data:
+            dictionary["content"] = form.cleaned_data["content"]
+        if "expiration" in form.cleaned_data:
+            dictionary["expiration"] = form.cleaned_data["expiration"]
+        if "staff_only" in form.cleaned_data:
+            dictionary["staff_only"] = form.cleaned_data["staff_only"]
+        return render(request, "kiosk/tool_post_comment.html", dictionary)
+
+    comment = form.save(commit=False)
+    comment.content = comment.content.strip()
+    comment.author = request.user
+    comment.expiration_date = (
+        None
+        if form.cleaned_data["expiration"] == -1
+        else timezone.now() + timedelta(days=form.cleaned_data["expiration"])
+    )
+    comment.save()
```

Comment:
not sure it's worth it here, but using a function from tool_control.py could be useful if the code is changed in the future